airflow/decorators/sensor.py

```diff
@@ -0,0 +1,80 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from inspect import signature
+from typing import Any, Callable, Collection, Dict, Mapping, Optional, Sequence
+
+from airflow.decorators.base import TaskDecorator, get_unique_task_id, task_decorator_factory
+from airflow.models.taskinstance import Context
+from airflow.sensors.base import BaseSensorOperator, PokeReturnValue
+
+
+class DecoratedSensorOperator(BaseSensorOperator):
```

Comment:
Why does this not use `PythonSensor`?

Reply:
Hi @uranusjr Thanks for the comments! The PR was blocked by CI tests failures which I don't think is related to the PR. Do you have any suggestions on this?

Reply:
I don't see not much benefit of having another layer of indirection.

Reply:
By not using `PythonSensor`, this implementation looses the ability to pass context variables (such as `ti`) to the callable. The ability of passing in `templates_dict` is also lost.